.github/workflows/ci.yml

```diff
@@ -0,0 +1,147 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Package version (e.g. v0.1.0)
+        required: false
+
+env:
+  # Common versions
+  GO_VERSION: '1.21.3'
+  GOLANGCI_VERSION: 'v1.54.2'
+  DOCKER_BUILDX_VERSION: 'v0.11.2'
+  XP_CHANNEL: master   # TODO(negz): Pin to stable once v1.14 is released.
+  XP_VERSION: current  # TODO(negz): Pin to a version once v1.14 is released.
+
+  # Common users. We can't run a step 'if secrets.USR != ""' but we can run
+  # a step 'if env.USR' != ""', so we copy these to succinctly test whether
+  # credentials have been provided before trying to run steps that need them.
+  XPKG_ACCESS_ID: ${{ secrets.XPKG_ACCESS_ID }}
```

Comment:
Nothing will be pushed to xpkg.upbound.io unless this secret (and the corresponding token secret) are created.

Reply:
Added (another) comment to clarify.